client/jira.go

```diff
@@ -129,6 +143,34 @@ func CreateIssue(project, issueType, title string) (domain.Issue, error) {
 	return issue, nil
 }
 
+func SearchIssues(query string, startIndex int, maxResults int) (SearchResult, error) {
+	var searchResult SearchResult
+
+	requestBody := newSearchRequestBody(query, startIndex, maxResults)
+	encodedBody, err := json.Marshal(requestBody)
+	if err != nil {
+		return searchResult, err
+	}
+
+	path := "/rest/api/2/search"
+	request, err := constructPostRequest(path, string(encodedBody))
+	if err != nil {
+		return searchResult, err
+	}
+	statusCode, err := SendRequest(request, &searchResult)
+
+	if statusCode != http.StatusOK {
+		err := fmt.Errorf("query='%s' : statusCode=%d", query, statusCode)
+		return searchResult, err
+	}
+	if err != nil {
```

Comment:
Should we move this error check before statusCode check?. Check `SendRequest` code, it looks like it could return a timeout error, and return a "-1". This "-1" is not something that a jira will return and the error message could be misleading. 
BTW, not sure why we are returning -1 but it's nothing related with this PR :)

Reply:
I've changed the order and added a test for it as well in b5302b9